pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -1405,3 +1407,62 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) {
 func (c *Controller) AppendWorkloadHandler(f func(*model.WorkloadInstance, model.Event)) {
 	c.handlers.AppendWorkloadHandler(f)
 }
+
+// stripPodUnusedFields is the transform function for shared pod informers,
+// it removes unused fields from objects before they are stored in the cache to save memory.
+func stripPodUnusedFields(obj any) (any, error) {
+	t, ok := obj.(metav1.ObjectMetaAccessor)
+	if !ok {
+		// shouldn't happen
+		return obj, nil
+	}
+	// ManagedFields is large and we never use it
+	t.GetObjectMeta().SetManagedFields(nil)
+	// Annotation is never used
+	t.GetObjectMeta().SetAnnotations(nil)
```

Comment:
Are we sure we will never use pod annotations in future also?

Reply:
At least not needed for now, not sure about future